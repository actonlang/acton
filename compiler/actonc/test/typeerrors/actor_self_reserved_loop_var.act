actor Test():
    for self in [1, 2, 3]:
        pass