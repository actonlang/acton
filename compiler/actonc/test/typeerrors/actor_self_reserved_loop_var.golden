Building file test/typeerrors/actor_self_reserved_loop_var.act using temporary scratch directory
  Compiling actor_self_reserved_loop_var.act for development
[error]: 'self' cannot be used as a parameter name in actors.
     +--> actor_self_reserved_loop_var.act@2:9-2:13
     |
   2 |     for self in [1, 2, 3]:
     :         ^--- 
     :         `- 'self' cannot be used as a parameter name in actors.
     :
     | Hint: In actors, 'self' is reserved and implicitly available as a reference to the own actor that can be passed to other actors.
-----+
